### PR TITLE
feat(speculative): add DeepSeek-V3 (MLA) EAGLE-3 draft model

### PR DESCRIPTION
**Pull request by khazic**
## What

Adds a DeepSeek-V3 (MLA) EAGLE-3 draft model
(`DeepseekV3Eagle3DraftModel` in `eagle/draft_deepseek.py`) and registers it for
`DeepseekV3ForCausalLM`.

## Why

The Llama-style EAGLE-3 draft (`LlamaEagle3DraftModel`) cannot represent
DeepSeek's **Multi-head Latent Attention**: queries/keys come from low-rank
`q_lora` / `kv_lora` projections, the rotary part is a separate `qk_rope_head_dim`
slice rotated with DeepSeek's *interleaved* RoPE, and the value head dim
(`v_head_dim`) differs from the q/k head dim. So DeepSeek targets get a dedicated
draft, following the same precedent as gpt-oss (`GptOssEagle3DraftModel`, which
got its own draft for a different RoPE).

## How

The draft mirrors `draft_llama` one-to-one for everything EAGLE-3 specific (the
`[embed, hidden]` fused first layer; the `cache_hidden = [K_list, V_list]` TTT
recurrence with per-step rotary phase offset and diagonal-extension attention;
`project_hidden_states` / `embed_input_ids` / `compute_logits` /
`set_vocab_mapping` and the `d2t` / `t2d` buffers) and only swaps the attention
for MLA. To keep the draft positionally consistent with the target, the MLA
projection layout and the interleaved RoPE are taken from the onboarded DeepSeek
target (`components/models/deepseek_v3`: the `MLA` projection structure and
`rope_utils`) rather than reimplemented. The shared `Eagle3TrainerModule` and
checkpointing are reused unchanged.

## Scope (v1)

EAGLE-3, eager attention, both Q paths (`q_lora_rank` set or `None`). P-EAGLE
parallel-drafting, flash-attention, and sequence packing are intentionally left
to follow-ups (they are orthogonal to the MLA attention this adds; `forward`
raises `NotImplementedError` for `seq_lens` packing).

## Validation

Architecture-level: unit tests cover forward shapes (both Q paths), the
`cache_hidden` TTT recurrence (including the diagonal-extension step), the
`d2t` / `t2d` vocab mapping, registry resolution, the packing guard, and **end-to-end
training through `Eagle3TrainerModule` on a tiny config (loss is finite and
gradients flow)**. A real acceptance benchmark needs a full DeepSeek target,
which is out of scope for CI; the rotary/projection math is anchored by reusing
the target's own `deepseek_v3` implementation.

### CODE REVIEW

File: nemo_automodel/components/speculative/eagle/draft_deepseek.py [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+        # Interleaved-RoPE frequencies, built exactly as the DeepSeek target does
+        # so the draft's rotary phase matches the target's.
+        freqs = precompute_freqs_cis(
+            self.qk_rope_head_dim,
+            getattr(config, "max_position_embeddings", 4096),
+            getattr(config, "rope_theta", 10000.0),
+            getattr(config, "rope_scaling", None),
+        )
+        self.register_buffer("rope_freqs", freqs, persistent=False)
```
[the remark this answers is not in the document]
Comment (khazic) [reply nemo_automodel/components/speculative/eagle/draft_deepseek.py:127]: done

**Review by HuiyingLi — APPROVED** [review]
